scripts/flowcell_summary_uppload_LIMS.py

```diff
@@ -42,68 +42,75 @@ def get_run_qcs(fc, lanesobj):
             #No demutiplexing found. this is fine.
             pass
 
-def  main(flowcell, all_flowcells,days,conf):
+def  main(flowcell, all_flowcells,days,conf,run_type):
     """If all_flowcells: all runs run less than a moth ago are uppdated"""
     today = date.today()
     couch = load_couch_server(conf)
     fc_db = couch['flowcells']
+    xfc_db = couch['x_flowcells']
+    process_dict = {'hiseq':'Illumina Sequencing (Illumina SBS) 4.0', 'miseq':'MiSeq Run (MiSeq) 4.0', 'hiseqx':'Illumina Sequencing (HiSeq X) 1.0'}    
+    # Collect flowcell processes based upon how script is called
     if all_flowcells:
-        flowcells = lims.get_processes(type = ['Illumina Sequencing (Illumina SBS) 4.0','MiSeq Run (MiSeq) 4.0'])
-        for fc in flowcells:
-            try:
-                closed = date(*map(int, fc.date_run.split('-')))
-                delta = today-closed
-            except AttributeError:
-                #Happens if fc has no date run, we should just not update and get to the next flowcell
+        flowcells = lims.get_processes(type = process_dict.values())
+    elif flowcell:
+        if not run_type:
+            raise SystemExit("Option -f should always be called with option -t. kindly refer -h fo rmore info.")        
+        fc_id = flowcell if run_type == 'miseq' else flowcell[1:]
+        try:
+            flowcells = [lims.get_processes(type = process_dict[run_type], udf = {'Flow Cell ID' : fc_id})[0]]
+            days = float('inf') #no need of days check when a flowcell is specified
+        except:
+            raise SystemExit("Could not find any process for FC %s (type: %s)" % (flowcell, run_type))
+    
+    for fc in flowcells:
+        try:
+            closed = date(*map(int, fc.date_run.split('-')))
+            delta = today-closed
+            if not delta.days < days:
```

Comment:
nope!! its the opposite. If the process is closed *greater* than specified days. 

Just realized could have simply written as `days > delta.days` instead of the `not ....` :sweat_smile: